unified_planning/solvers/parallel.py

```diff
@@ -56,25 +58,74 @@ def _run_parallel(self, fname, *args):
                                signaling_queue, fname, *args))
             processes.append(_p)
             _p.start()
+        processes_alive = len(processes)
+        results: List[up.solvers.PlanGenerationResult] = []
+        definitive_result_found: bool = False
+        timeout: bool = False
         while True:
+            if processes_alive == 0: # Every planner gave a result
+                break
             (idx, res) = signaling_queue.get(block=True)
+            processes_alive -= 1
             if isinstance(res, BaseException):
                 raise res
             else:
-                break
+                assert isinstance(res, up.solvers.PlanGenerationResult)
+                # If the planner is sure about the result (optimality of the result or impossibility of the problem) exit the loop
+                if res.status == up.solvers.results.SOLVED_OPTIMALLY or res.status == up.solvers.results.UNSOLVABLE_PROVEN:
+                    definitive_result_found = True
+                    break
+                else:
+                    if res.plan is not None:
+                        results.append(res)
+                    if res.status == up.solvers.results.TIMEOUT:
+                        timeout = True
         for p in processes:
             p.terminate()
-        return res
+        if definitive_result_found: # A planner found a definitive result
+            return res
+        else:
+            result_order: List[int] = [ up.solvers.results.SOLVED_SATISFICING,  # List containing the results in the order we prefer them
+                                        up.solvers.results.TIMEOUT,
+                                        up.solvers.results.UNSOLVABLE_INCOMPLETELY,
+                                        up.solvers.results.MEMOUT,
+                                        up.solvers.results.INTERNAL_ERROR]
+            for ro in result_order:
+                for r in results:
+                    if r.status == ro:
+                        return r
+            # if no results are given by the planner, we create a default one
+            return up.solvers.PlanGenerationResult(up.solvers.results.TIMEOUT if timeout else up.solvers.results.UNSOLVABLE_INCOMPLETELY,
```

Comment:
Here I would keep only UNSOLVABLE_INCOMPLETELY

Reply:
Why? If a planner went in TIMEOUT it would make sense to let the user know that with more time a plan might be found, while UNSOLVABLE_INCOMPLETELY means that, even with more time, those planners will not find a plan.

Reply:
This is not solved. You need to remove the timeout.